src/pages/recipes/detail.js

```diff
@@ -0,0 +1,95 @@
+import { ClockIcon, BellIcon, LinkIcon, FireIcon } from "@heroicons/react/24/outline";
+import { useEffect } from "react";
+
+const Modal = ({ isOpen, onClose, recipe }) => {
+    useEffect(() => {
+        if (isOpen) {
+            const handleEsc = (event) => {
+                if (event.key === "Escape") {
+                    onClose();
+                }
+            };
+            window.addEventListener("keydown", handleEsc);
+            return () => {
+                window.removeEventListener("keydown", handleEsc);
+            };
+        }
+    }, [isOpen, onClose]);
+
+    if (!isOpen || !recipe) return null;
+
+    return (
+        <div 
+            className="fixed inset-0 bg-gray-900 bg-opacity-90 flex items-center justify-center z-50 p-4 sm:p-8"
+            role="dialog"
+            aria-modal="true"
+        >
+            <div className="bg-gray-800 text-gray-200 rounded-lg max-w-full sm:max-w-lg w-full max-h-[90vh] overflow-y-auto relative shadow-xl p-6">
+                <button
+                    className="absolute top-2 right-4 text-gray-300 hover:text-gray-100 transition duration-300 text-2xl"
+                    onClick={onClose}
+                    aria-label="Close modal"
+                >
+                    &times;
+                </button>
+
+                <h2 className="text-lg sm:text-xl md:text-2xl lg:text-3xl text-emerald-400 text-center font-bold mb-4 mt-6">
+                    {recipe.name}
+                </h2>
+
+                <div className="flex justify-center mb-4 p-4">
+                    <img
+                        src={recipe.image} 
+                        alt={recipe.name}
+                        className="max-w-full h-auto rounded-lg border border-gray-600 shadow-md"
+                    />
```

Comment:
hi @SeesonLau, using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images.